test/test_torch.py

```diff
@@ -13367,6 +13367,35 @@ def test_cat_big(self, device):
         result = torch.cat(concat_list)
         self.assertEqual(result.size(0), SIZE1 + SIZE2)
 
+    @onlyCUDA
+    def test_atomic_add(self, device):
+        # todo: we should probably expose a dtype.is_signed()
+        def is_signed(dtype):
+            return torch.is_signed(torch.tensor(0, dtype=dtype))
+        # https://github.com/pytorch/pytorch/issues/29153
+        for dtype in torch.testing.get_all_math_dtypes(device):
+            size = [5, 5]
+            if dtype.is_floating_point:
+                tensor = torch.rand(size, dtype=dtype, device=device)
+            if is_signed(dtype):
+                tensor = torch.randint(-5, 15, size, dtype=dtype, device=device)
+            else:
+                tensor = torch.randint(0, 10, size, dtype=dtype, device=device)
+
+            # index_add calls atomicAdd on cuda.
```

Comment:
It seems like we're relying on `index_add` calling atomicAdd. It might be clearer to introduce a C++ API test that only calls atomicAdd so that we're still testing this behavior if someone changes the implementation of `index_add`

Reply:
I'll check with you tomorrow. Last I looked it didn't seem like we had the infrastructure setup to build and run tests on cu files. If you know an easy way to do that or where they'd go maybe point me to it, otherwise I think setting that up could be a separate task.

Reply:
Some of our cpp tests definitely run on CUDA: https://github.com/pytorch/pytorch/blob/master/test/cpp/jit/test_fuser.cpp